snapcraft/internal/mangling.py

```diff
@@ -0,0 +1,48 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+
+from snapcraft import file_utils
+
+
+def rewrite_python_shebangs(root_dir):
+    """Recursively change #!/usr/bin/pythonX shebangs to #!/usr/bin/env pythonX
```

Comment:
Shouldn't this live in the Python plugin? Although on second thought I guess we wouldn't want repo to be depending on plugin code...

Reply:
Nah, several things besides the plugin needs to use it, as you can see.